pwiz_tools/Skyline/FileUI/PeptideSearch/ImportPeptideSearchDlg.cs

```diff
@@ -909,7 +909,10 @@ private void AddDetectedFeaturesToDocument(IProgressMonitor progressMonitor)
             // Add the library molecules to the document, in natural sort order
             var status = new ProgressStatus(PeptideSearchResources.ImportPeptideSearchDlg_NextPage_Adding_detected_features_to_document);
             progressMonitor.UpdateProgress(status);
-            Assume.AreNotEqual(Document.Settings.PeptideSettings.Libraries.LibrarySpecs, _existingLibraries.LibrarySpecs);
+            if (Equals(Document.Settings.PeptideSettings.Libraries.LibrarySpecs, _existingLibraries.LibrarySpecs))
+            {
+                MessageDlg.Show(this, PeptideSearchResources.ImportPeptideSearchDlg_AddDetectedFeaturesToDocument_Did_not_detect_any_features___check_your_settings_and_try_again);

```

Comment:
This seems strange that there is no return in this if block. The if is quick enough that it seems like you should do it ahead of the progress update and then just return. If there is some reason you need to update the status, that should be noted and then add the line to complete the status and return. It doesn't make sense to go through all of the code below, and it puts a burden on anyone changing it to ensure it still works for this degenerate case.